### PR TITLE
[rmqcpp] Add port for rmqcpp - a C++ RabbitMQ Client Library

### PR DESCRIPTION
**Pull request by ahsanabbas123**
### Summary 
Adding a port for `rmqcpp`, a C++ RabbitMQ Client Library open-sourced by Bloomberg - https://github.com/bloomberg/rmqcpp/tree/main. 

### Checklist
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/rmqcpp/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "rmqcpp",
+  "version": "1.0.0",
+  "description": "rmqcpp - A C++ library for RabbitMQ.",
+  "dependencies": [
+    {
+      "name": "bde",
```
Comment (FrankXie05) [inline ports/rmqcpp/vcpkg.json:7]: Note:`bde` provided by `vcpkg` supports limited triplets.

https://github.com/microsoft/vcpkg/blob/3715d743ac08146d9b7714085c1babdba9f262d5/ports/bde/vcpkg.json#L5

**Comment by ahsanabbas123** [comment]
Hi @FrankXie05, anything else we need to address here or is it good to approve?

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
@ahsanabbas123 Test on `x64-linux` error: 
```
 
-- Building x64-linux-dbg
-- Building x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /mnt/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake --build . --config Release -- -v -j9
    Working Directory: /mnt/vcpkg/buildtrees/rmqcpp/x64-linux-rel
    See logs for more information:
      /mnt/vcpkg/buildtrees/rmqcpp/build-x64-linux-rel-out.log
```
Error:
```
mnt/vcpkg/buildtrees/rmqcpp/src/3213daa1b6-c9f1ce5735/src/rmq/rmqamqpt/rmqamqpt_types.cpp: In member function ‘void BloombergLP::rmqamqpt::FieldValueEncoder::operator()(float) const’:
/mnt/vcpkg/buildtrees/rmqcpp/src/3213daa1b6-c9f1ce5735/src/rmq/rmqamqpt/rmqamqpt_types.cpp:372:14: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  372 |             *reinterpret_cast<const bsl::uint32_t*>(&val);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/vcpkg/buildtrees/rmqcpp/src/3213daa1b6-c9f1ce5735/src/rmq/rmqamqpt/rmqamqpt_types.cpp: In member function ‘void BloombergLP::rmqamqpt::FieldValueEncoder::operator()(double) const’:
/mnt/vcpkg/buildtrees/rmqcpp/src/3213daa1b6-c9f1ce5735/src/rmq/rmqamqpt/rmqamqpt_types.cpp:379:14: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
  379 |             *reinterpret_cast<const bsl::uint64_t*>(&val);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by ahsanabbas123** [comment]
Thanks for pointing out the `x64-linux-rel` errors @FrankXie05, the build should work now after the 2 fixes referenced above - is it good to go now?

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
The usage has been tested successfully locally.

File: ports/rmqcpp/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "rmqcpp",
+  "version": "1.0.0",
```
Comment (dg0yt) [inline ports/rmqcpp/vcpkg.json:3]: Does this package really have a version "1.0.0", or should the port use `version-date`?

**Comment by JavierMatosD** [comment]
The port LGTM, but upstream, doesn't have a "1.0.0" release. @ahsanabbas123, are you planning on doing a "1.0.0" release anytime soon? If not, we should switch the version scheme to `version-date` as @dg0yt suggested above.

File: ports/rmqcpp/vcpkg.json (hunk 2 above)
Comment (ahsanabbas123) [reply ports/rmqcpp/vcpkg.json:3]: Released `1.0.0`

**Comment by ahsanabbas123** [comment]
> The port LGTM, but upstream, doesn't have a "1.0.0" release. @ahsanabbas123, are you planning on doing a "1.0.0" release anytime soon? If not, we should switch the version scheme to `version-date` as @dg0yt suggested above.

Thanks for pointing that out - released `1.0.0`

**Comment by olejniczak** [comment]
Hi @ahsanabbas123 
I try to build bde with vcpkg but become build error:

`vcpkg.exe install bde:x64-windows --allow-unsupported
Computing installation plan...
warning: bde is only supported on '!windows & !arm & !android & !osx', which does not match x64-windows. This usually means that there are known build failures, or runtime problems, when building other platforms. Proceeding anyway due to `--allow-unsupported`.
The following packages will be built and installed:
    bde:x64-windows@3.124.0.0
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\oley\AppData\Local\vcpkg\archives in 825 us. Use --debug to see more details.
Installing 1/1 bde:x64-windows@3.124.0.0...
Building bde:x64-windows@3.124.0.0...
-- Note: bde only supports static library linkage. Building static library.
-- Using cached bloomberg-bde-tools-3.124.0.0.tar.gz.
-- Cleaning sources at D:/src/libs/vcpkg_rmqcpp/buildtrees/bde/src/3.124.0.0-8b4e52b527.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/src/libs/vcpkg_rmqcpp/downloads/bloomberg-bde-tools-3.124.0.0.tar.gz
-- Using source at D:/src/libs/vcpkg_rmqcpp/buildtrees/bde/src/3.124.0.0-8b4e52b527.clean
-- Configure bde-tools-v
-- Using cached bloomberg-bde-3.124.0.0.tar.gz.
-- Cleaning sources at D:/src/libs/vcpkg_rmqcpp/buildtrees/bde/src/3.124.0.0-e9e36169e4.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/src/libs/vcpkg_rmqcpp/downloads/bloomberg-bde-3.124.0.0.tar.gz
-- Using source at D:/src/libs/vcpkg_rmqcpp/buildtrees/bde/src/3.124.0.0-e9e36169e4.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
CMake Warning at installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:318 (message):
  The following variables are not used in CMakeLists.txt:

      BDE_BUILD_TARGET_DBG

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  ports/bde/portfile.cmake:30 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Building x64-windows-dbg
-- Building x64-windows-rel
CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.27/Modules/GNUInstallDirs.cmake:243 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  ports/bde/portfile.cmake:52 (include)
  scripts/ports.cmake:170 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Installing: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/share/bde/copyright
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/bal.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/bbl.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/bdl.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/bsl.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/inteldfp.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/pcre2.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/ryu.pc
-- Fixing pkgconfig file: D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig/s-baltst.pc
-- Using cached msys2-mingw-w64-x86_64-pkgconf-1~2.1.0-1-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.10-4-x86_64.pkg.tar.zst.
-- Using msys root at D:/src/libs/vcpkg_rmqcpp/downloads/tools/msys2/fdbea3694fb5c0d4
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:141 (message):

  D:/src/libs/vcpkg_rmqcpp/downloads/tools/msys2/fdbea3694fb5c0d4/mingw64/bin/pkg-config.exe
  --exists bal failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/lib/pkgconfig;D:/src/libs/vcpkg_rmqcpp/packages/bde_x64-windows/share/pkgconfig;D:/src/libs/vcpkg_rmqcpp/installed/x64-windows/lib/pkgconfig;D:/src/libs/vcpkg_rmqcpp/installed/x64-windows/share/pkgconfig"
      output: Package Ws2-32 was not found in the pkg-config search path.

  Perhaps you should add the directory containing `Ws2-32.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'Ws2-32', required by 'bsl', not found

  Package 'Ws2-32', required by 'bsl', not found
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:206 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/bde/portfile.cmake:63 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:170 (include)


error: building bde:x64-windows failed with: BUILD_FAILED
Elapsed time to handle bde:x64-windows: 3.7 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+bde
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[bde]+Build+error+on+x64-windows&body=Copy+issue+body+from+D%3A%2Fsrc%2Flibs%2Fvcpkg_rmqcpp%2Finstalled%2Fvcpkg%2Fissue_body.md
`

without it I'm not able to build rmqcpp, which I want to try.

**Comment by FrankXie05** [comment]
@olejniczak Plese open a new issue. :)

**Pull request merged**